fabric-rendering-v1/src/client/java/net/fabricmc/fabric/impl/client/rendering/tooltip/MultiTooltipData.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.client.rendering.tooltip;
+
+import java.util.ArrayList;
+
+import net.minecraft.client.item.TooltipData;
+
+/**
+ * This class stores multiple TooltipData object to their further mapping to MultiTooltipComponent.
+ */
+public class MultiTooltipData extends ArrayList<TooltipData> implements TooltipData {
```

Comment:
Extending ArrayList feels like an unusual design pattern, something like this + a separate array list passed to it  would probably be more conventional:
```java
public record MultiTooltipData(List<TooltipData> children) implements TooltipData {}
```

Reply:
Yeah, i was concerned about this too, thanks for the solution